include/nbl/builtin/glsl/blit/blit/blit.glsl

```diff
@@ -0,0 +1,177 @@
+#ifndef _NBL_GLSL_BLIT_INCLUDED_
+#define _NBL_GLSL_BLIT_INCLUDED_
+
+#ifndef _NBL_GLSL_BLIT_MAIN_DEFINED_
+
+#include <nbl/builtin/glsl/blit/parameters.glsl>
+nbl_glsl_blit_parameters_t nbl_glsl_blit_getParameters();
+
+vec4 nbl_glsl_blit_getData(in vec3 texCoord, in uint layerIdx);
+void nbl_glsl_blit_setData(in vec4 data, in uvec3 coord, in uint layerIdx);
+vec4 nbl_glsl_blit_getKernelWeight(in uint index);
+void nbl_glsl_blit_addToHistogram(in uint bucketIndex, in uint layerIdx);
+
+#ifndef _NBL_GLSL_BLIT_ALPHA_BIN_COUNT_
+#error _NBL_GLSL_BLIT_ALPHA_BIN_COUNT_ must be defined
+#endif
+
+#define scratchShared _NBL_GLSL_SCRATCH_SHARED_DEFINED_
+
+uvec3 linearIndexTo3DIndex(in uint linearIndex, in uvec3 gridDim)
```

Comment:
This is basically `nbl_glsl_multi_dimensional_array_addressing_snakeCurveInverse` and should be called so.

Reply:
would be useful to have an overload impl
```glsl
uvec3 nbl_glsl_multi_dimensional_array_addressing_snakeCurveInverse (in uint linearIndex, in uvec2 gridDimPrefixProduct)
{
        uvec3 index3D;

	index3D.x = linearIndex / gridDimPrefixProduct.x;
	const uint tmp = linearIndex - index3D.z * gridDimPrefixProduct.y;
	index3D.y = tmp / gridDimPrefixProduct.x;
	index3D.x = tmp - index3D.y*gridDimPrefixProduct.x;
	
	return index3D;
}

uvec3 nbl_glsl_multi_dimensional_array_addressing_snakeCurveInverse (in uint linearIndex, in uvec3 gridDim)
{
    return nbl_glsl_multi_dimensional_array_addressing_snakeCurveInverse(linearIndex,uvec2(gridDim.x,gridDim.x*gridDim.y));
}
```

Reply:
Resolved. But probably can't supply prefix products from push constants now.

Reply:
once you start providing the `uvec2`, some variables might be unused.

Also maximum image sizes and array layers counts are always <=32k on all platforms, so can be packed into 16 bits

Reply:
Resolved.